pkg/fqdn/name_manager.go

```diff
@@ -139,11 +145,21 @@ func NewNameManager(config Config) *NameManager {
 		}
 	}
 
+	// Setup the update queue and a trivial goroutine to apply them
+	updateQueue := make(chan func(), 1024)
```

Comment:
Should we test this with an unbuffered channel to make sure we don't deadlock when the buffer gets full?